site/documentation/content/api/telemetry-pubsub/index.md

```diff
@@ -0,0 +1,118 @@
+1. The client is authorized to write to the topic.
+1. The device for which the adapter wants to send telemetry data has been registered (see
+   [Device Registration API]({{< relref "/api/device-registration" >}})).
+
+**Message Flow**
+
+It is up to the discretion of the protocol adapter whether it wants to use *AT LEAST ONCE* or *AT MOST ONCE* delivery
+semantics.
+
+The following sequence diagram illustrates the flow of messages involved in the *HTTP Adapter* publishing a telemetry
+data message to the Pub/Sub API implementing *AT MOST ONCE* delivery semantics.
+
+{{< figure src="publish_pubsub_qos0.svg" title="Publish telemetry data flow (AT MOST ONCE)" >}}
+
+1. *Device* `4711` PUTs telemetry data to the *HTTP Adapter*
+    1. *HTTP Adapter* publishes telemetry data to the *Pub/Sub API*.
+    1. *HTTP Adapter* acknowledges the reception of the data to the *Device*.
+
+The following sequence diagram illustrates the flow of messages involved in the *HTTP Adapter* publishing a telemetry
+data message to the Pub/Sub API implementing *AT LEAST ONCE* delivery semantics.
+
+{{< figure src="publish_pubsub_qos1.svg" title="Publish telemetry data flow (AT LEAST ONCE)" >}}
+
+1. *Device* `4711` PUTs telemetry data to the *HTTP Adapter*, indicating *QoS Level* 1.
+    1. *HTTP Adapter* publishes telemetry data to the *Pub/Sub API*.
+    1. *Pub/Sub* acknowledges reception of the message.
+    1. *HTTP Adapter* acknowledges the reception of the data to the *Device*.
+
+When a Pub/Sub publisher raises an exception while sending a telemetry message to Pub/Sub, the protocol adapter MUST NOT try
+to re-send such rejected messages but SHOULD indicate the failed transfer to the device if the transport protocol
+provides means to do so.
+
+**Message Format**
+
+The following tables provides an overview of the message format for a PubsubMessage. The message must contain either a non-empty data field or at least one attribute.
+
+| Name            | Type        | Description |
+| :-------------- | :---------- | :---------- |
+| *data*  | *string*    | The message data field. If this field is empty, the message must contain at least one attribute. |
+| *attributes* | *map*      | Attributes for this message. If this field is empty, the message must contain non-empty data. This can be used to filter messages on the subscription. |
+| *messageId* | *string*      | ID of this message, assigned by the server when the message is published. Guaranteed to be unique within the topic.  |
+| *publishTime* | *string*      | The time at which the message was published, populated by the server when it receives the publish call. |
+| *orderingKey* | *string*      | If non-empty, identifies related messages for which publish order should be respected. If a Subscription has enableMessageOrdering set to true, messages published with the same non-empty orderingKey value will be delivered to subscribers in the order in which they are received by the Pub/Sub system. |
```

Comment:
IMHO it would make sense to define that a client (protocol adapter) MUST put the device ID into this field, wouldn't it? That would be in analogy to how we defined that the Kafka message key must contain the device id ...